src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,233 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import '@itwin/itwinui-css/css/date-picker.css';
+
+const isSameHour = (hours: number, date: Date | undefined) => {
+  return !!date && hours === date.getHours();
+};
+
+const areSameMinutes = (minutes: number, date: Date | undefined) => {
+  return !!date && minutes === date.getMinutes();
+};
+
+const areSameSeconds = (seconds: number, date: Date | undefined) => {
+  return !!date && seconds === date.getSeconds();
+};
+
+const isSamePeriod = (period: string, date: Date | undefined) => {
+  return (
+    !!date && (period === 'AM' ? date.getHours() < 12 : date.getHours() >= 12)
+  );
+};
+
+export type TimePickerProps = {
+  /**
+   * Selected date.
+   */
+  date?: Date;
+  /**
+   * Callback when date is changed.
+   */
+  onChange?: (date: Date) => void;
+  /**
+   * Format of the time. 12h or 24h.
+   * @default false
+   */
+  use12Hours?: boolean;
+  /**
+   * Precision of the time.
+   * @default 'minutes'
+   */
+  precision?: 'hours' | 'minutes' | 'seconds';
+  /**
+   * Change step of the hours displayed.
+   * @default 1
+   */
+  hourStep?: number;
+  /**
+   * Change step of the minutes displayed.
+   * @default 1
+   */
+  minuteStep?: number;
+  /**
+   * Change step of the seconds displayed.
+   * @default 1
+   */
+  secondStep?: number;
+};
+
+/**
+ * Time picker component
+ * @example
+ * <TimePicker date={new Date()} onChange={(e) => console.log('New time value: ' + e)} />
+ */
+export const TimePicker = (props: TimePickerProps): JSX.Element => {
+  const {
+    date,
+    onChange,
+    use12Hours = false,
+    precision = 'minutes',
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+  } = props;
+
+  useTheme();
+
+  const [selectedTime, setSelectedTime] = React.useState(date);
```

Comment:
So it would always have the default value of current time. I don't think this should be a problem because you currently use current time everywhere.

Reply:
I have selectedDate in DatePicker and there it can be undefined.. So I either leave it like that or change in both places then..

Reply:
I think here might be a different scenario because if no date is provided and you select hours then minutes and seconds gets auto selected. Maybe we should initialize with `00:00:00` also in datepicker if date is not provided then set time with zeros.